### PR TITLE
fix finalise sale bug and add validations as per certik audit

### PR DESCRIPTION
**Pull request by Divyansh-Gupta-Antier**

### CODE REVIEW

**Review by dzikowski — APPROVED** [review]

File: chaincode/src/utils/dexUtils.ts [hunk 1]
```diff
@@ -110,3 +117,13 @@ export async function fetchDexProtocolFeeConfig(ctx: GalaChainContext): Promise<
...
   return dexConfig;
 }
+
+/**
+ * @dev it will check whether tickLower is smaller than tickUpper in input
+ * @param tickLower
+ * @param tickUpper
+ * @throw error if condition is invalid
+ */
+export function areTicksValid(tickLower: number, tickUpper: number) {
```
[the remark this answers is not in the document]
Comment (Divyansh-Gupta-Antier) [reply chaincode/src/utils/dexUtils.ts]: Updated the code to use a custom validator instead of the method. Updated e2e test cases to work with these. Will add top level test cases for all functions with edge cases like this one in a future PR.
